docs/developer/using-makefiles.md

```diff
@@ -0,0 +1,58 @@
+# How to set up the environment using the provided makefile targets
+
+The main file is [Makefile](https://github.com/kubeapps/kubeapps/blob/master/Makefile), which will compile and prepare the production assets for then generating a set of Docker images. It is the starting point when you want to build the Kubeapps different components.
+
+For setting up the environment for running Kubeapps, we also provide (as is) makefile targets for:
+* Creating a multicluster environment with Kind ([cluster-kind.mk](https://github.com/kubeapps/kubeapps/blob/master/script/cluster-kind.mk))
+* Deploying and configuring the components for getting Kubeapps running with OIDC login using Dex ([deploy-dev.mk](https://github.com/kubeapps/kubeapps/blob/master/script/deploy-dev.mk)).
+
+> Disclaimer: these files are not being actively maintained, as they are solely intended for helping Kubeapp developers to set up the environment. If you are a contributor and you are having troubles, please feel free to [open an issue](https://github.com/kubeapps/kubeapps/issues/new).
+
+## Makefile for generating images
+
+### Commands
+
+Find below a list of the most used commands:
+
+``` bash
+make # will make all the kubeapps images
+make kubeapps/dashboard
+make kubeapps/apprepository-controller
+make kubeapps/kubeops
+make kubeapps/assetsvc
+make kubeapps/asset-syncer
+```
+
+> You can set the image tag manually: `IMAGE_TAG=myTag make`
+
+## Makefile for setting up the environment
+
+### Prerequisites
+
+* Install `mkcert`; you can get it from the [official repository](https://github.com/FiloSottile/mkcert/releases).
+* Get the Kind network IP and replace it when necessary.
+    * Retrieve the IP for Dex:
+    It is a requirement to discover the nodes IP address on the bridge network so that Dex can be reached both inside and outside the cluster at the same address. 
+    You can get this IP by inspecting the kind network (`docker network inspect kind`) and setting the value as **the next available IP on that network**.
+    
+        * Another way to do so is to execute `kubectl --namespace=kube-system get pods -o wide | grep kube-apiserver-kubeapps-control-plane  | awk '{print $6}'`, but it won't work unless you exec `make delete-cluster-kind && make cluster-kind`, as some of these files (the apiserver-config ones) are config for the cluster apiserver itself, which has to know where to find dex.
```

Comment:
Just to add a little more clarity: "Another way to do so is to start the environment with `make cluster-kind` and manually verify the IP address by executing `kubectl ....`, but you will need to re-create the cluster after you've updated the config files (below) by executing `make delet....`, as some ...."